### DISCUSSION TITLE
Improve push notifications

### DISCUSSION THREAD

**Discussion by ozturkberkay** [tutao/tutanota#5037] [Ideas]
Hi,

Unfortunately, email notifications from Tutanota are not really useful right now. Without seeing the email subject (and/or the sender email address), there is no way of knowing if the notification is worth your attention. Since loading the emails takes like 3 seconds on my phone, it becomes very time consuming to open the app for email notifications that are not urgent. To be honest, this is the only reason I haven't upgraded to the Premium plan yet. I've been checking the roadmap for the last few months. I find it a bit concerning that this is still not included in the roadmap.

There are two open issues for this:
- [Android](https://github.com/tutao/tutanota/issues/4542)
- [iOS](https://github.com/tutao/tutanota/issues/3668)

Are there any plans for improving the notifications?

**Comment by charlag** [comment]
Hi, this is definitely something we should improve on, if not subject (might also be possible but has risks) then at least the sender address would be very desirable. Issues are open which means we want to implement it.

**Reply by ozturkberkay** [reply]
Sender address would be enough for me. Thanks for the acknowledgment, I'll keep my eyes on the roadmap. Hopefully this will prioritized sometime soon.

**Comment by coxde** [comment]
It might be a good idea to provide options in the settings for the user to decide whether to show the message in the notification/show the sender/show the message subject/both.

**Comment by snaggen** [comment]
As a premium user, the useless notifications are actually a reason I'm looking into other mail providers. 
I use Tutanota as my primary mail provider, hence almost all my incoming mails are from regular insecure mail. Hence, they are already compromised from a security perspective. Hence, it wouldn't affect my security as a whole if I got an option to include the sender/subject in the notifications. As is now, I only get disturbed by the notifications, but I do not bother to act on them since it is slow to open the app and the notification is most likely not important (since I use it as main mail, I get lots of non important mails also). Turning notifications into just an annoyance, I have actually thought about disable notifications for Tutanota for this reason but it feels a bit radical.
